README.md

```diff
@@ -1,2 +1,422 @@
+            accessibility: .afterFirstUnlock
+        )
+    }
+
+    subscript<CodableType: Codable>(_ key: String, ofType type: CodableType.Type) -> CodableType? {
+        guard let data = try? object(forKey: key) else { return nil }
+        return try? JSONDecoder().decode(type, from: data)
+    }
+
+    subscript<CodableType: Codable>(key: String) -> CodableType? {
+        get {
+            self[key, ofType: CodableType.self]
+        }
+        set {
+            if let newValue, let data = try? JSONEncoder().encode(newValue) {
+                try? setObject(data, forKey: key)
+            } else {
+                try? removeObject(forKey: key)
+            }
+        }
+    }
+}
+```
+
+### @Instantiated
+
+Property declarations within [`@Instantiable`](#instantiable) types decorated with the [`@Instantiated` macro](Sources/SafeDI/PropertyDecoration/Instantiated.swift) are instantiated when its enclosing type is instantiated. `@Instantiated`-decorated properties are available to be [`@Received`](#received) by objects instantiated further down the dependency tree.
+
+`@Instantiated`-decorated properties must declared as an `@Instantiable` type, or of an `additionalType` listed in a `@Instantiable(fulfillingAdditionalTypes:)`‘s declaration.
+
+#### Utilizing @Instantiated with type erased properties
+
+When you want to instantiate a type-erased property, you may specify which concrete type you expect to fulfill your property by utilizing `@Instantiated`‘s `fulfilledByType` parameter.
+
+```swift
+import SwiftUI
+
+@Instantiable
+public struct ParentView: View {
+    public var body: some View {
+        VStack {
+            Text("Child View")
+            childViewBuilder.instantiate()
+        }
+    }
+
+    public init(childViewBuilder: Instantiator<some View>) {
+        self.childViewBuilder = childViewBuilder
+    }
+
+    // The Instantiator‘s `instantiate()` method will build a view of type `ChildView`.
+    // Because the type is passed in as a string literal, this code does not need to
+    // have a dependency on the module that defines `ChildView`. All that is required
+    // for this code to compile is for there to be an
+    // `@Instantiable public struct ChildView: View` in the codebase.
+    @Instantiated(fulfilledByType: "ChildView")
+    private let childViewBuilder: Instantiator<some View>
+}
+```
+
+### @Forwarded
+
+Property declarations within [`@Instantiable`](#instantiable) types decorated with the [`@Forwarded` macro](Sources/SafeDI/PropertyDecoration/Forwarded.swift) are forwarded into the SafeDI dependency tree by a [`ForwardingInstantiator`](Sources/SafeDI/DelayedInstantiation/ForwardingInstantiator.swift) instance’s `instantiate(…)` method. A `@Forwarded`-decorated property is available to be [`@Received`](#received) by objects instantiated further down the dependency tree.
+
+A single `@Instantiable` type may have at most one `@Forwarded`-decorated property.
```

Comment:
I think we probably should provide a matrix illustrating the situations when one should use instantiated, forwarded, and received. It would make it clear at a glance what to use in what scenarios

Reply:
Extremely smart. We're in need of better visual organization of this README for sure, and a table would be a solid step forward here.

Reply:
Added a "macro cheat sheet". The table in this section copied wording from elsewhere but I think the table representation is still helpful as an easy reference. LMK what you think.